src/Aspire.Hosting.Azure.CognitiveServices/IHasDeploymentsResource.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure OpenAI resource.
+/// </summary>
+public interface IHasDeploymentResource: IResourceWithConnectionString, IResourceWithEnvironment
```

Comment:
This is a surprisingly general type name - `Aspire.Hosting.ApplicationModel.IHasDeploymentResource` doesn't sound specific to Azure OpenAI but from the implementation it is. Could this be named something more specific to AzureOpenAI?